src/amo/components/AddonQRCodeLink/index.js

```diff
@@ -0,0 +1,114 @@
+/* @flow */
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import config from 'config';
+
+import { CLIENT_APP_ANDROID } from 'amo/constants';
+import AddonQRCode from 'amo/components/AddonQRCode';
+import Button from 'amo/components/Button';
+import OverlayCard from 'amo/components/OverlayCard';
+import translate from 'amo/i18n/translate';
+import type { AppState } from 'amo/store';
+import type { AddonType } from 'amo/types/addons';
+import type { ElementEvent, HTMLElementEventHandler } from 'amo/types/dom';
+import type { I18nType } from 'amo/types/i18n';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType,
+|};
+
+export type DeafultProps = {|
+  _config: typeof config,
+|};
+
+type PropsFromState = {|
+  clientApp: string,
+|};
+
+export type InternalProps = {|
+  ...Props,
+  ...DeafultProps,
+  ...PropsFromState,
+  i18n: I18nType,
+|};
+
+type State = {|
+  showQRCode: boolean,
+|};
+
+export class AddonQRCodeLinkBase extends React.Component<InternalProps, State> {
+  static defaultProps: DeafultProps = {
+    _config: config,
+  };
+
+  constructor(props: InternalProps) {
+    super(props);
+
+    this.state = {
+      showQRCode: false,
+    };
+  }
+
+  onShowQRCode: HTMLElementEventHandler = (event: ElementEvent) => {
+    event.preventDefault();
+
+    this.setState({ showQRCode: true });
+  };
+
+  onHideQRCode: () => void = () => {
+    this.setState({ showQRCode: false });
+  };
+
+  render(): React.Node | null {
+    const { _config, addon, clientApp, i18n } = this.props;
+
+    if (
+      !_config.get('addonIdsWithQRCodes').includes(addon.id) ||
+      clientApp === CLIENT_APP_ANDROID ||
+      !_config.get('enableFeatureAddonQRCode')
+    ) {
+      return null;
```

Comment:
Given that this is done here, you could also make `addon` an optional prop and check it here. That would allow the caller to not have to worry about `addon` being defined.

Reply:
Sure, I can do that.